### ISSUE TITLE
TaintSyncPackage does not load correctly in Visual Studio 2019

### ISSUE THREAD

**Issue by rlusian1** [SonarSource/sonarlint-visualstudio#2287]
### Description

We have multiple developers who are getting error messages about the TaintSyncPackage not being able to load:

---------------------------
Microsoft Visual Studio
---------------------------
The 'TaintSyncPackage' package did not load correctly.
The problem may have been caused by a configuration change or by the installation of another extension. You can get more information by examining the file 'C:\Users\itron\AppData\Roaming\Microsoft\VisualStudio\16.0_c7f9311d\ActivityLog.xml'.

This is an excerpt from an ActivityLog.xml file:

 <entry>
    <record>971</record>
    <time>2021/04/05 16:34:01.235</time>
    <type>Information</type>
    <source>VisualStudio</source>
    <description>Begin package load [TaintSyncPackage]</description>
    <guid>{EEEB81FA-D6C3-438D-B29E-9FAAC91471CC}</guid>
  </entry>
  <entry>
    <record>972</record>
    <time>2021/04/05 16:34:01.265</time>
    <type>Error</type>
    <source>VisualStudio</source>
    <description>SetSite failed for package [TaintSyncPackage]Source: &apos;SonarLint.VisualStudio.IssueVisualization.Security&apos; Description: Object reference not set to an instance of an object.&#x000D;&#x000A;System.NullReferenceException: Object reference not set to an instance of an object.&#x000D;&#x000A;   at SonarLint.VisualStudio.IssueVisualization.Security.Taint.TaintIssuesSynchronizer.IsFeatureSupported()&#x000D;&#x000A;   at SonarLint.VisualStudio.IssueVisualization.Security.Taint.TaintIssuesSynchronizer.&lt;SynchronizeWithServer&gt;d__10.MoveNext()&#x000D;&#x000A;--- End of stack trace from previous location where exception was thrown ---&#x000D;&#x000A;   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)&#x000D;&#x000A;   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)&#x000D;&#x000A;   at SonarLint.VisualStudio.IssueVisualization.Security.Taint.TaintSyncPackage.&lt;InitializeAsync&gt;d__1.MoveNext()&#x000D;&#x000A;--- End of stack trace from previous location where exception was thrown ---&#x000D;&#x000A;   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)&#x000D;&#x000A;   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)&#x000D;&#x000A;   at Microsoft.VisualStudio.Shell.AsyncPackage.&lt;&gt;c__DisplayClass16_0.&lt;&lt;Microsoft-VisualStudio-Shell-Interop-IAsyncLoadablePackageInitialize-Initialize&gt;b__1&gt;d.MoveNext()&#x000D;&#x000A;--- End of stack trace from previous location where exception was thrown ---&#x000D;&#x000A;   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)&#x000D;&#x000A;   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)&#x000D;&#x000A;   at Microsoft.VisualStudio.Threading.JoinableTask.&lt;JoinAsync&gt;d__70.MoveNext()</description>
    <guid>{EEEB81FA-D6C3-438D-B29E-9FAAC91471CC}</guid>
    <hr>80004003 - E_POINTER</hr>
    <errorinfo></errorinfo>
  </entry>
  <entry>
    <record>973</record>
    <time>2021/04/05 16:34:01.275</time>
    <type>Error</type>
    <source>VisualStudio</source>
    <description>End package load [TaintSyncPackage]</description>
    <guid>{EEEB81FA-D6C3-438D-B29E-9FAAC91471CC}</guid>
    <hr>80004003 - E_POINTER</hr>
    <errorinfo></errorinfo>
  </entry>

### Repro steps

Please provide the steps required to reproduce the problem

1. Add the Sonar Lint for Visual Studio 2019 extension.
2. Start Visual Studio. It doesn't happen every time, however.

### Expected behavior

TaintSyncPackage should load without error.

### Actual behavior

See above.

### Known workarounds

Don't know of any ATM.

### Related information

* Sonar Lint for Visual Studio 2019, version 4.33.1.28913.
* Visual Studio Enterprise 2019: 16.9.2 and 16.8.6
* SonarQube server version: (don't have this info at the moment)
* Languages in your solution: C#

**Comment by rita-sonarsource** [comment]
Hi @rlusian1, thanks for reporting this to us, we'll look into it. Are there any logs in the `Output Window -> SonarLint` pane?

**Comment by ronnydahl** [comment]
I had the same issue until I found out I wasn't connected to the same network that our Sonar server runs on. As soon as I joined the correct network (note: by VPN in my case with git-proxy) and restarted Visual Studio, everything worked as expected. 

Here is my logs:

Connecting to 'https://xxx/'.
Using default daemon download URL
Daemon version: 4.3.2.2521
Daemon download url: https://binaries.sonarsource.com/Distribution/sonarlint-daemon/sonarlint-daemon-4.3.2.2521-windows.zip
Current VS version:
    Visual Studio Professional 2019
    16.9.31205.134
    16.9.4
Settings file does not exist at "C:\Users\xxx\AppData\Roaming\SonarLint for Visual Studio\settings.json".
Default settings will be used
Checking for suppressions...
SonarQube request failed: An error occurred while sending the request. Unable to connect to the remote server
Failed to fetch suppressions: An error occurred while sending the request.
Initializing the telemetry package...
Finished initializing the telemetry package...
Initializing the SonarLint package...
Connected mode detected.
Finishing initializing the SonarLint package
Initializing the notifications package...
Loading notifications settings...
Connected: checking for notifications
Finished initializing the notifications package
[Taint] Initializing taint issues synchronization package...
Failed to fetch notifications: An error occurred while sending the request.
Failed to fetch notifications: An error occurred while sending the request.
Initializing the daemon package...
[Open in IDE] Creating request listener...
[Open in IDE] Checking availability of port 64120
Finished initializing the daemon package.
[Open in IDE] Request listener created successfully. Listening on port 64120.
